src/GraphQL/InputMapper/Search/Criterion/ParentLocationId.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\Criterion;
+
+use eZ\Publish\API\Repository\Values\Content\Query;
+use EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\QueryBuilder;
+use EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\QueryInputVisitor;
+
+class ParentLocationId implements QueryInputVisitor
+{
+    public function visit(QueryBuilder $queryBuilder, $value): void
+    {
+        $queryBuilder->addCriterion(new Query\Criterion\ParentLocationId($value));
```

Comment:
Should we refactor all visitors that instanciate a Criterion object to a generic one ? It could take a criterion class as a constructor argument. We could remove a few duplicate files with this.

Reply:
but Field and DateMetadata seems to work a bit different than the others, aren't they?

Reply:
Indeed !

But you could at least remove `ContentTypeId`, `ContentTypeIdentifier`, `ParentLocationId` and `Text`. Furthermore, there are more criteria that work the same way, we could easily add them if we have a class for it.

`BasicCriterionVisitor` ? `Standard` ?

Reply:
i don't know if i follow you here. 

do you mean having a Standard criterion visitor tagged with allo those input keys? 
my doubt is, how to instatiate this standard criterion and tell it to use one criterion or another depending on the inputkey ?

Reply:
An example:

```
class BasicCriterionVisitor implements QueryInputVisitor
{
    public function __construct($criterionClass)
    {
        $this->criterionClass = $criterionClass;
    }

    public function visit(QueryBuilder $queryBuilder, $value): void
    {
        $queryBuilder->addCriterion(new {$this->criterionClass}($value));
```

```
services:
    EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\Criterion\ContentTypeId:
        class: EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\Criterion\Basic
        arguments:
            $criterionClass: 'eZ\Publish\API\Repository\Values\Content\Query\Criterion\ContentTypeId'
        tags:
            - {name: 'ezplatform_graphql.query_input_visitor', inputKey: 'ContentTypeId'}    

    EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\Criterion\ContentTypeIdentifier:
        class: EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\Criterion\Basic
        arguments:
            $criterionClass: 'eZ\Publish\API\Repository\Values\Content\Query\Criterion\ContentTypeIdentifier'
        tags:
            - {name: 'ezplatform_graphql.query_input_visitor', inputKey: 'ContentTypeIdentifier'}    

EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\Criterion\ParentLocationId:
        class: EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\Criterion\Basic
        arguments:
            $criterionClass: 'eZ\Publish\API\Repository\Values\Content\Query\Criterion\ParentLocationId'
        tags:
            - {name: 'ezplatform_graphql.query_input_visitor', inputKey: 'ParentLocationId'}    
```

Does it make more sense ?

Reply:
yes, much more clear now. i thought you were trying to avoid all those service definitions, but you're only trying to avoid php classes :)
i'll work on this.